homeassistant/components/sia/config_flow.py

```diff
@@ -0,0 +1,173 @@
+"""Config flow for sia integration."""
+import logging
+
+from pysiaalarm import (
+    InvalidAccountFormatError,
+    InvalidAccountLengthError,
+    InvalidKeyFormatError,
+    InvalidKeyLengthError,
+    SIAAccount,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PORT
+
+from .const import (  # pylint: disable=unused-import
+    CONF_ACCOUNT,
+    CONF_ACCOUNTS,
+    CONF_ADDITIONAL_ACCOUNTS,
+    CONF_ENCRYPTION_KEY,
+    CONF_PING_INTERVAL,
+    CONF_ZONES,
+    DOMAIN,
+    INVALID_ACCOUNT_FORMAT,
+    INVALID_ACCOUNT_LENGTH,
+    INVALID_KEY_FORMAT,
+    INVALID_KEY_LENGTH,
+    INVALID_PING,
+    INVALID_ZONES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def validate_input(data: dict) -> dict:
+    """Validate the input by the user."""
+    errors = {}
+    if data:
+        try:
+            SIAAccount(data[CONF_ACCOUNT], data.get(CONF_ENCRYPTION_KEY))
+        except InvalidKeyFormatError:
+            errors["base"] = INVALID_KEY_FORMAT
+        except InvalidKeyLengthError:
+            errors["base"] = INVALID_KEY_LENGTH
+        except InvalidAccountFormatError:
+            errors["base"] = INVALID_ACCOUNT_FORMAT
+        except InvalidAccountLengthError:
+            errors["base"] = INVALID_ACCOUNT_LENGTH
+        except Exception:  # pylint: disable=broad-except
+            errors["base"] = "unknown"
+
+        if not 1 <= int(data[CONF_PING_INTERVAL]) <= 1440:
+            errors["base"] = INVALID_PING
+        if int(data[CONF_ZONES]) <= 0:
+            errors["base"] = INVALID_ZONES
+    return errors
+
+
+def create_schema(step_id: str, user_input: dict, errors: dict) -> vol.Schema:
+    """Create a schema for the next or first form, using the errors to delete erroneous input, but filling the other fields."""
```

Comment:
You don't need to delete erroneous input - let the user fix it.

Reply:
good idea, made it a bit simpler as well